packages/plugins/response-cache/src/plugin.ts

```diff
@@ -684,3 +690,26 @@ export const cacheControlDirective = /* GraphQL */ `
 
   directive @cacheControl(maxAge: Int, scope: CacheControlScope) on FIELD_DEFINITION | OBJECT
 `;
+
+function removeMetadataFieldsFromResult(data: Record<string, unknown>): any {
+  if (Array.isArray(data)) {
+    return data.map(removeMetadataFieldsFromResult);
+  }
+  // clone the data to avoid mutation
+  data = { ...data };
+
+  if (data.__responseCacheTypeName) {
+    delete data.__responseCacheTypeName;
+  }
+  if (data.__responseCacheId) {
+    delete data.__responseCacheId;
+  }
```

Comment:
It is safe to delete things even if they don't exists
```suggestion
  delete data.__responseCacheTypeName;
  delete data.__responseCacheId;
```